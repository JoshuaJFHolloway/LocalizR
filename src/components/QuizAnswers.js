import React, { Component } from 'react';
import SubmitButton from './SubmitButton'

  class QuizAnswers extends Component {
    constructor(){
      super();
    }

    randomiser = (array= [1,2,3,4]) => {
      for (var i = array.length - 1; i > 0; i--) {
        var j = Math.floor(Math.random() * (i + 1));
        [array[i], array[j]] = [array[j], array[i]];
      }
        var num1 = array[0];
        return [num1];
      };

    decider = () => {
      var values = this.randomiser();
        if(values[0] === 1){
          return 'Selection1'
        } if(values[0] === 2){
          return 'Selection2'
      } if(values[0] === 3){
        return 'Selection3'
      } if(values[0] === 4){
        return 'Selection4'
      }
    };

    render() {

      const Views = {
        Selection1: (
          <div>
            <SubmitButton
              buttonName={this.props.answer1}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer2}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer3}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer4}
              handleDataSubmitted={this.props.handleCorrectAnswerSubmitted}
            />
          </div>
        ),

        Selection2: (
          <div>
            <SubmitButton
              buttonName={this.props.answer1}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer2}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer4}
              handleDataSubmitted={this.props.handleCorrectAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer3}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
          </div>
        ),

        Selection3: (
          <div>
            <SubmitButton
              buttonName={this.props.answer1}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer4}
              handleDataSubmitted={this.props.handleCorrectAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer2}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer3}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
          </div>
        ),

        Selection4: (
          <div>
            <SubmitButton
              buttonName={this.props.answer4}
              handleDataSubmitted={this.props.handleCorrectAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer1}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer2}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
            <SubmitButton
              buttonName={this.props.answer3}
              handleDataSubmitted={this.props.handleAnswerSubmitted}
            />
          </div>
        )
      };

      return (
        <div>
          {Views[this.decider()]}
        </div>
      )
    };
}

export default QuizAnswers;
