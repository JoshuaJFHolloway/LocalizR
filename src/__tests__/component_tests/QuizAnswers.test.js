import React from 'react';
import { shallow, mount } from 'enzyme';
import QuizAnswers from '../../components/QuizAnswers';

describe('QuizAnswers', () =>{

  let quizanswers = shallow(<QuizAnswers answer={'el vestido rojo'} />)

  describe('renders components correctly',() =>{
    it('renders correctly', () => {
      expect(quizanswers).toMatchSnapshot();
    });

    it('renders QuizAnswer', () =>{
     expect(quizanswers.find('QuizAnswer').exists()).toBe(true);
    });

    it('renders QuizCorrectAnswer', () => {
      expect(quizanswers.find('QuizCorrectAnswer').exists()).toBe(true);
    });

    it('renders 3 QuizAnswer components', () => {
      expect(quizanswers.find('QuizAnswer').length).toEqual(1);
    });

  });
});
