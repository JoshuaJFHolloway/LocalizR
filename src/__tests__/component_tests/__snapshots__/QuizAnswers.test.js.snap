// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizAnswers renders components correctly renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuizAnswers />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <QuizAnswer
          answer="el vestido rojo"
          handleAnswerSubmitted={undefined}
/>,
        <QuizCorrectAnswer
          answer="el traje rojo"
          handleCorrectAnswerSubmitted={undefined}
/>,
        <QuizAnswer
          answer="el abrigo rojo"
          handleAnswerSubmitted={undefined}
/>,
        <QuizAnswer
          answer="los pantalones rojos"
          handleAnswerSubmitted={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "answer": "el vestido rojo",
          "handleAnswerSubmitted": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "answer": "el traje rojo",
          "handleCorrectAnswerSubmitted": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "answer": "el abrigo rojo",
          "handleAnswerSubmitted": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "answer": "los pantalones rojos",
          "handleAnswerSubmitted": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <QuizAnswer
            answer="el vestido rojo"
            handleAnswerSubmitted={undefined}
/>,
          <QuizCorrectAnswer
            answer="el traje rojo"
            handleCorrectAnswerSubmitted={undefined}
/>,
          <QuizAnswer
            answer="el abrigo rojo"
            handleAnswerSubmitted={undefined}
/>,
          <QuizAnswer
            answer="los pantalones rojos"
            handleAnswerSubmitted={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "answer": "el vestido rojo",
            "handleAnswerSubmitted": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "answer": "el traje rojo",
            "handleCorrectAnswerSubmitted": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "answer": "el abrigo rojo",
            "handleAnswerSubmitted": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "answer": "los pantalones rojos",
            "handleAnswerSubmitted": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
